packages/cheqd-sdk/src/dids/CheqdDidRegistrar.ts

```diff
@@ -0,0 +1,423 @@
+  DidDeactivateResult,
+  DidUpdateResult,
+  DidDocument,
+  Jwk,
+  VerificationMethod,
+} from '@aries-framework/core'
+import type { CheqdNetwork, DIDDocument, DidStdFee, TVerificationKey, VerificationMethods } from '@cheqd/sdk'
+import type { SignInfo } from '@cheqd/ts-proto/cheqd/did/v2'
+
+import {
+  DidDocumentRole,
+  DidRecord,
+  DidRepository,
+  KeyType,
+  Key,
+  Buffer,
+  isValidPrivateKey,
+} from '@aries-framework/core'
+import { MethodSpecificIdAlgo, createDidVerificationMethod } from '@cheqd/sdk'
+import { MsgCreateResourcePayload } from '@cheqd/ts-proto/cheqd/resource/v2'
+import { fromString, toString } from 'uint8arrays'
+import { v4 } from 'uuid'
+
+import { CheqdSdkLedgerService } from '../ledger'
+
+import {
+  MsgCreateDidDocPayloadToSign,
+  generateDidDoc,
+  validateSpecCompliantPayload,
+  fromMultibase,
+  MsgDeactivateDidDocPayloadToSign,
+} from './didCheqdUtil'
+
+export class CheqdDidRegistrar implements DidRegistrar {
+  public readonly supportedMethods = ['cheqd']
+
+  public async create(agentContext: AgentContext, options: CheqdDidCreateOptions): Promise<DidCreateResult> {
+    const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+    const cheqdSdkLedgerService = agentContext.dependencyManager.resolve(CheqdSdkLedgerService)
+
+    const { methodSpecificIdAlgo, network, versionId = v4() } = options.options
+    const { verificationMethod } = options.secret
+    let didDocument: DidDocument
+
+    if (options.didDocument && validateSpecCompliantPayload(options.didDocument)) {
+      didDocument = options.didDocument
+    } else if (verificationMethod) {
+      const privateKey = verificationMethod.privateKey
+      if (privateKey && !isValidPrivateKey(privateKey, KeyType.Ed25519)) {
+        return {
+          didDocumentMetadata: {},
+          didRegistrationMetadata: {},
+          didState: {
+            state: 'failed',
+            reason: 'Invalid private key provided',
+          },
+        }
+      }
+
+      const key = await agentContext.wallet.createKey({
+        keyType: KeyType.Ed25519,
+        privateKey: privateKey,
+      })
+
+      didDocument = generateDidDoc({
+        verificationMethod: verificationMethod.type,
+        verificationMethodId: verificationMethod.id || 'key-1',
+        methodSpecificIdAlgo: methodSpecificIdAlgo || MethodSpecificIdAlgo.Uuid,
+        network,
+        publicKey: toString(key.publicKey, 'base64'),
+      }) as DidDocument
+    } else {
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'failed',
+          reason: 'Provide a didDocument or atleast one verificationMethod with seed in secret',
+        },
+      }
+    }
+
+    try {
+      const payloadToSign = await MsgCreateDidDocPayloadToSign(didDocument as DIDDocument, versionId)
+      const signInputs = await this.signPayload(agentContext, payloadToSign, didDocument.verificationMethod!)
```

Comment:
Even though we know the did document will have the verification method,  I think it would be good to handle the case of it not existing. If the `generateDidDoc` method changes we won't get any errors here of it not existing.

You can also make the return type of `generateDidDoc` to return `DidDocument & { verificationMethod: VerificadtionMethod[] }` so it assures the vMethod is there